gobblin-metrics/src/main/java/gobblin/metrics/influxdb/InfluxDBReporter.java

```diff
@@ -121,50 +125,133 @@ public static Builder builder(String name) {
     return new Builder(name);
   }
 
-  private Serie.Builder addGauge(String name, Gauge gauge, long timeStamp, Serie.Builder builder) {
-    return builder.values(timeStamp, name, gauge.getValue());
+  private void reportGauge(MetricContext context, String name, Gauge gauge, long timeStamp) {
+    this.influxDB.write(
```

Comment:
Does writing happen asynchronously? Otherwise it seems like the overhead of creating HTTP requests will dominate any other performance gain.
You could return a list of series from each of these methods, and then push either as a single batch, or as multiple batches of a few series each.